lib/puppet_x/msutter/templates/invoke_dsc_resource.erb

```diff
@@ -4,33 +4,10 @@ if (!(Test-Path($PuppetModulesFolder))) {
   & cmd.exe /c mklink /d "$PuppetModulesFolder" "<%= vendored_modules_path %>" | Out-Null
 }
 
-function Set-RefreshMode
-{
-  param($mode)
-
-  $configOutputPath = [IO.Path]::GetTempPath()
-
-  [DscLocalConfigurationManager()]
-  Configuration LCMSettings {
-      Node localhost
-      {
-          Settings
-          {
-              RefreshMode = $mode
-          }
-      }
-  }
-
-  LCMSettings -OutputPath $configOutputPath | Out-Null
-
-  Set-DscLocalConfigurationManager -Path $configOutputPath
-}
-
 $currentState = Get-DscLocalConfigurationManager
 
 if ($currentState.RefreshMode -ne 'Disabled') {
-  $currentRefreshMode = $currentState.RefreshMode
-  Set-RefreshMode 'Disabled'
+  throw "DSC LCM RefreshMode must be set to Disabled for Puppet to execute DSC Resources!"
```

Comment:
I have confirmed that we probably want to improve this error reporting, because the way it bails right now is pretty user-unfriendly.  Certainly good enough for this PR, but we have room for improvement in a future iteration.


```
Error: /Stage[main]/Main/Dsc_registry[registry_test]: Could not evaluate: Execution of 'C:\Windows\sysnative\WindowsPowershell\v1.0\powershell.exe -NoProfile -NonInteractive -NoLogo -ExecutionPolicy Bypass -Command $script:ErrorActionPreference = 'Stop'
$PuppetModulesFolder = Join-Path $env:ProgramFiles 'WindowsPowerShell\Modules\PuppetVendoredModules'
if (!(Test-Path($PuppetModulesFolder))) {
  & cmd.exe /c mklink /d "$PuppetModulesFolder" "c:\source\dsc\puppetlabs-dsc\spec\fixtures\modules\dsc\vendor\DSCResources" | Out-Null
}

$currentState = Get-DscLocalConfigurationManager

if ($currentState.RefreshMode -ne 'Disabled') {
  throw "DSC LCM RefreshMode must be set to Disabled for Puppet to execute DSC Resources!"
}

$invokeParams = @{
  Name          = 'Registry'
  Method        = 'test'
  ErrorAction   = 'Continue'
  ErrorVariable = 'invokeError'
  Property      = @{
    key = 'HKEY_LOCAL_MACHINE\SOFTWARE\TestKey'
    valuename = 'TestStringValue'
    valuedata = @('Cats')
    valuetype = 'String'
    ensure = 'Present'
  }
  ModuleName = "PSDesiredStateConfiguration"
}

$result = Invoke-DscResource @invokeParams

if ($currentRefreshMode) { Set-RefreshMode $currentRefreshMode }
if ($invokeError) { throw $invokeError }

switch ($invokeParams.Method) {
  'Test' {
    [bool]($result.InDesiredState)
  }
  'Set' {
    @($Null, 0) -contains $result.ReturnValue
  }
}
' returned 1: DSC LCM RefreshMode must be set to Disabled for Puppet to execute DSC Resources!
At line:10 char:3
+   throw "DSC LCM RefreshMode must be set to Disabled for Puppet to ex ...
+   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (DSC LCM Refresh... DSC Resources!:String) [], R
   untimeException
    + FullyQualifiedErrorId : DSC LCM RefreshMode must be set to Disabled for Puppet to execute
   DSC Resources!
```